main.py

```diff
@@ -0,0 +1,51 @@
+originArray = [[1,2,3],[4,5,6],[7,8,9]]
+finalArray = []
+
```

Comment:
Most of the following 4 functions in the following lines look pretty much the same.
Can you somehow merge them into one small function? 
Try to be creative. If you feel like you ruin your code, try to think about another way.